### PR TITLE
Time - Mair H

### PR DESCRIPTION
**Pull request by mheshmati-tech**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | From my understanding Models are responsible for maintaining/manipulating data. |
Describe in your own words what the Controller is doing in Rails | The controller is where actions and the logic resides- for instance for this assignment, each action found the task by their id the applied the desired changes using the logic in place. It is the middle man between Model and View, talking to the data base and the way the data needs to be displayed to the user. |
Describe in your own words what the View is doing in Rails | View is the html or in other words what the user sees/front end portion of Rails. |
Describe an edge-case controller test you wrote | What program should do given an id that doesn't exist in our database. |
What is the purpose of using strong params? (i.e. the params method in the controller) | To DRY up the code, this reduced the lines of code tremendously. |
How are Rails migrations related to Rails models? | Based off of my understanding, they're a way to manipulate the data- every time a migration happens the schema is updated starting with nothing each time. Active Record keeps track of the changes based off the timeline. |
Describe one area of Rails that are still unclear on | database migration, Model, and Routes somewhat. |

### CODE REVIEW

File: db/migrate/20200429234130_create_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateTasks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tasks do |t|
+      t.string :name
+      t.string :description
+      t.string :completed_at
```
Comment (kaidamasaki) [inline db/migrate/20200429234130_create_tasks.rb:6]: In the future you should use `t.datetime` for columns like this.

**Comment by kaidamasaki** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | Unstyled views.
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements | ✔️
| Yellow (Approaches Standards) | 3+ in Code Review && 4+ in Functional Requirements, or the instructor judges that this project needs special attention |
| Red (Not at Standard) | 0-2 in Code Review or 0-3 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Perfect Indentation | ✅
| Elegant/Clever | ✅
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅

**Review by kaidamasaki — COMMENTED** [review]
Great job!  Just have one tip for future projects.  😃
